Tba/Moves the API to ES5.x

Tobsucht opened the pull request:
- moves the API to ES 5.x
- adds support for span queries
- adds test cases for aggregations and span queries
- adds regex mode for the completion suggester

Tobsucht committed:
Bump up elastic search version to 5.1.2

Tobsucht committed:
add log4j, remove embedded node support

Tobsucht committed:
Fix index-prefix and scripts

Tobsucht committed:
Remove redundant dependency

Tobsucht committed:
Remove redundant Node class

Tobsucht committed:
Remove redundant imports, increase version of log4j

Tobsucht committed:
Adjust completion suggester API, test cases

Tobsucht committed:
Completion suggester defaults to no context usage

Tobsucht committed:
Since es-5.x a nested query needs a score mode

Tobsucht committed:
Remove analyzer from object field as this is not allowed

andyHa committed:
Cherry-pick changes from old branch

Tobsucht committed:
Cherry pick nested fields

Tobsucht committed:
remove test-scope

Tobsucht committed:
Fixes a NPE

Tobsucht committed:
Name for groups of filters

Tobsucht committed:
Ordering of term aggregations

Tobsucht committed:
Regex mode for the completion suggester

Tobsucht committed:
Adds support for near span queries

Tobsucht committed:
Update pom

Tobsucht committed:
Merge branch 'master' into tba/es-5.x

# Conflicts:
#	pom.xml
#	src/main/java/sirius/search/constraints/NestedField.java

Tobsucht committed:
Fixes a faulty merge

andyHa commented on src/main/java/sirius/search/IndexLifecycle.java:
```diff
@@ -31,7 +36,7 @@ public int getPriority() {
     public void started() {
         if (Strings.isEmpty(Sirius.getConfig().getString("index.type"))) {
-            IndexAccess.LOG.INFO("ElasticSearch is disabled! (index.type is not set)");
+            LOG.INFO("ElasticSearch is disabled! (index.type is not set)");
```
brauchen wir index-type überhaupt noch?

andyHa commented on src/main/java/sirius/search/DateRange.java:
```diff
@@ -212,17 +213,25 @@ protected String getKey() {
+                                                               .toEpochMilli()));
             }
         } else if (from == null) {
             if (until != null) {
```
always true

andyHa commented on src/main/java/sirius/search/IndexAccess.java:
```diff
@@ -515,34 +457,17 @@ private void updateMappings() {
-                Exceptions.handle(LOG, e);
-            }
-            client = transportClient;
+
```
warning wenn in index.type was anderes als host drin steht oder so

Tobsucht committed:
Refactors span constraints

Tobsucht committed:
Removed index.type, code style

andyHa committed:
Support for creating and deleting temporary indices.

As ES no longer supports in memory instances, we now resort temporarily created indices

andyHa committed:
Merge branch 'tba/es-5.x' of github.com:scireum/sirius-search into tba/es-5.x

# Conflicts:
#	src/main/java/sirius/search/IndexAccess.java
#	src/main/resources/component-search.conf
#	src/test/resources/component-test-search.conf

Tobsucht committed:
Remove unused imports

andyHa committed:
Cleanup and minor fixes

andyHa committed:
Merge remote-tracking branch 'origin/tba/es-5.x' into tba/es-5.x

# Conflicts:
#	src/main/java/sirius/search/IndexLifecycle.java

andyHa committed:
Permits to disable es by setting an empty host in the config
